crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -782,29 +782,24 @@ impl<'a> BackwardsTokenizer<'a> {
         self.back_offset = self.cursor.text_len() + self.offset;
 
         if self.after_newline {
-            // This comment ended a line with a higher line number, not the current one
-            let previous_comment_idx = self.previous_comment_idx.unwrap_or_else(|| {
-                self.comment_ranges
-                    .partition_point(|comment| comment.end() <= self.back_offset)
-            });
-            // If `previous_comment_idx == 0`, we're in a comment free region
-            if previous_comment_idx > 0 {
-                let comment = self.comment_ranges[previous_comment_idx - 1];
-                if comment.end() == self.back_offset {
-                    // Skip the comment without iterating over the chars manually
-                    self.cursor =
-                        Cursor::new(&self.source[TextRange::new(self.offset, comment.start())]);
-                    debug_assert_eq!(self.cursor.text_len() + self.offset, comment.start());
-                    self.after_newline = false;
-                    self.previous_comment_idx = Some(previous_comment_idx - 1);
-                    return SimpleToken {
-                        kind: SimpleTokenKind::Comment,
-                        range: comment.range(),
-                    };
-                }
-                // At least memoize the binary search
-                self.previous_comment_idx = Some(previous_comment_idx);
+            if let Some(comment) = self
```

Comment:
Nit: May be worth testing if the `after_newline` boolean still is an optimization, considering that testing whether we're inside of a comment is now a single operation (unlike before where it was necessary to lex-out the full line, worst case, the entire file up to this position)